src/storage/buffer_manager.cpp

```diff
@@ -48,7 +49,7 @@ unique_ptr<BufferHandle> BlockHandle::Load(shared_ptr<BlockHandle> &handle) {
 		// this is mainly down to the block manager only having a single pointer into the file
 		// this is relatively easy to fix later on
 		lock_guard<mutex> buffer_lock(buffer_manager.manager_lock);
```

Comment:
Silly question: Does the lock need to cover the `Read`?

Reply:
I believe the comment right above that should answer this :)

```
// FIXME: currently we still require a lock for reading blocks from disk
// this is mainly down to the block manager only having a single pointer into the file
// this is relatively easy to fix later on
```

This should indeed be fixed at some point. I am also not sure about file handles there, perhaps on Windows we will need to have multiple file handles for multi-threaded reads as well.

Reply:
Sorry, I didn't see the comment in the diff.

Reply:
No worries, just thought it was amusing :)